src/ServiceManagement/Compute/Commands.ServiceManagement.Preview/Automation/RunCodeGeneration.ps1

```diff
@@ -1331,23 +1413,77 @@ function Write-OperationCmdletFile
             $param_definition = $indents + "public ${paramTypeNormalizedName} ${normalized_param_name} " + $get_set_block + $NEW_LINE;
             $invoke_param_definition = $indents + "public ${paramTypeNormalizedName} ${invoke_param_set_name}${normalized_param_name} " + $get_set_block + $NEW_LINE;
             $param_index = $position_index - 1;
-            $invoke_local_param_definition = $indents + (' ' * 4) + "${paramTypeNormalizedName} " + $pt.Name + " = (${paramTypeNormalizedName})ParseParameter(${invoke_input_params_name}[${param_index}]);";
-            $create_local_param_definition = $indents + (' ' * 4) + "${paramTypeNormalizedName} " + $pt.Name + " = ${param_constructor_code};";
+
+            if ($only_strings)
+            {
+                 $invoke_local_param_definition = $indents + (' ' * 4) + "var " + $pt.Name + " = new ${paramTypeNormalizedName}();" + $NEW_LINE;
+
+                 foreach ($param in $does_contain_only_strings)
+                 {
+                      $invoke_local_param_definition += $indents + (' ' * 4) + "var p${param} = (string) ParseParameter(${invoke_input_params_name}[${param_index}]);" + $NEW_LINE;
+                      $invoke_local_param_definition += $indents + (' ' * 4) + $pt.Name + ".${param} = string.IsNullOrEmpty(p${param}) ? null : p${param};" + $NEW_LINE;
+                      $param_index += 1;
+                      $position_index += 1;
+                 }
+            }
+            elseif ($has_properties -and ($is_string_list -eq $null))
+            {
+                 $invoke_local_param_definition = $indents + (' ' * 4) + "${paramTypeNormalizedName} " + $pt.Name + " = (${paramTypeNormalizedName})ParseParameter(${invoke_input_params_name}[${param_index}]);" + $NEW_LINE;
+            }
+            elseif ($has_properties)
+            {
+                 $invoke_local_param_definition = $indents + (' ' * 4) + "var inputArray${param_index} = Array.ConvertAll((object []) ParseParameter(${invoke_input_params_name}[${param_index}]), e => e.ToString());" + $NEW_LINE;
+                 $invoke_local_param_definition += $indents + (' ' * 4) + "${paramTypeNormalizedName} " + $pt.Name + "  = new ${paramTypeNormalizedName}();" + $NEW_LINE;
+                 $invoke_local_param_definition += $indents + (' ' * 4) + $pt.Name + ".${is_string_list} = inputArray${param_index}.ToList();" + $NEW_LINE;
+            }
+            else
+            {
+                 $invoke_local_param_definition = $indents + (' ' * 4) + "${paramTypeNormalizedName} " + $pt.Name + " = new ${paramTypeNormalizedName}();" + $NEW_LINE;
+            }
+
+
+            if ($only_strings)
```

Comment:
The condition check may need to include ```CLI``` so that the generator won't affect the CLI code (yet).